source/staking/contracts/Staking.sol

```diff
@@ -85,57 +74,69 @@ contract Staking is Ownable {
   }
 
   /**
-   * @notice Stake tokens
-   * @param amount uint256
+   * @notice Add delegate for account
+   * @param delegate address
    */
-  function stake(uint256 amount) external {
-    stakeFor(msg.sender, amount);
+  function addDelegate(address delegate) external {
+    require(!isDelegate[delegate], "ALREADY_DELEGATE");
+    require(allStakes[delegate].balance == 0, "ALREADY_STAKING");
+    accountDelegate[msg.sender] = delegate;
```

Comment:
This will set `delegate` whether that account wishes to be set as a delegate. Recommend that we include some kind of propose-accept flow to protect from mistakes and potential abuse.

Reply:
Created a new function called proposeDelegate, which is called by an account, and now setDelegate is called & confirmed by the proposed delegater